src/nvim/message.c

```diff
@@ -466,15 +466,7 @@ int emsg_not_now(void)
   return FALSE;
 }
 
-/*
- * emsg() - display an error message
- *
- * Rings the bell, if appropriate, and calls message() to do the real work
- * When terminal not initialized (yet) mch_errmsg(..) is used.
- *
- * return TRUE if wait_return not called
- */
-int emsg(const char_u *s_)
+static int _emsg(const char *s_, int (*msg_attr)(const char *s, const int attr))
```

Comment:
It is bad idea to mask symbols; also need typedef, AFAIR C declarations generator intentionally cannot swallow that.

Reply:
I made the typedef in a commit but I did not understood the "mask symbol" part (I mean, technically what does that mean? I'm sorry, my knowledge of C is not that advanced).

Reply:
It means that you should not define a symbol in some inner scope which is the same as a symbol in an outer scope. E.g. in

```C
int i = 0;
{
  int i = 1;
  printf("%d\n", i);
}
printf("%d\n", i);
```

second `i` masks the first one. `msg_attr` is a function defined in compilation unit scope, and at the same time an argument to `_emsg` which means that it is redefined in `_emsg` functions scope.

Reply:
Indeed it was not smart, thanks for the explanation!